StripeCore/StripeCore/Source/Analytics/STPAnalyticsClient.swift

```diff
@@ -115,11 +115,9 @@ import UIKit
         delegate?.analyticsClientDidLog(analyticsClient: self, payload: payload)
         #endif
 
+        // Don't send the analytic, but add it to `_testLogHistory` if we're in a test.
         guard type(of: self).shouldCollectAnalytics() else {
-            // Don't send the analytic, but add it to `_testLogHistory` if we're in a test.
-            if NSClassFromString("XCTest") != nil {
-                _testLogHistory.append(payload)
-            }
+            _testLogHistory.append(payload)
```

Comment:
nit: I don't think it's right to assume that `shouldCollectAnalytics` tells us whether or not we're in a test, so we should keep the explicit XCTest check.

Reply:
Chatted in Slack, confirmed that this doesn't change how we populate testmode or is_developement.
1. testmode is derived from the publishable key
2. is_development is derived from STPAnalyticsClient.isSimulatorOrTest